Compile warnings from sys/atomic.h

Thank you for taking the time to submit an issue!

## Background information

Warnings showing up during compile

### What version of the PMIx Reference Library are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)

git master @ 5d96ca1441acde9313e36ed8d9198066095985b3

### Describe how PMIx was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

clone

### Please describe the system on which you are running

* Operating system/version: CentOS 7.5
* Computer hardware: x86
* Network type: IB

-----------------------------

## Details of the problem

Warning messages during compile (per each use of sys/atomic.h).  Compilation succeeds though.

```
  CC       pmix_mca_base_components_register.lo
In file included from /gpfs/projects/ChapmanGroup/src/openpmix/pmix/pmix-git/src/threads/thread_usage.h:31:0,
                 from /gpfs/projects/ChapmanGroup/src/openpmix/pmix/pmix-git/src/class/pmix_object.h:131,
                 from /gpfs/projects/ChapmanGroup/src/openpmix/pmix/pmix-git/src/class/pmix_list.h:77,
                 from ../../../../pmix-git/src/mca/base/pmix_mca_base_components_register.c:30:
/gpfs/projects/ChapmanGroup/src/openpmix/pmix/pmix-git/src/atomics/sys/atomic.h:167:32: warning: "PMIX_BUILTIN_SYNC" is not defined [-Wundef]
 #elif PMIX_ASSEMBLY_BUILTIN == PMIX_BUILTIN_SYNC
                                ^
```

--- a/src/atomics/sys/atomic.h
+++ b/src/atomics/sys/atomic.h
@@ -164,8 +164,6 @@ enum {
  *********************************************************************/
 #if defined(DOXYGEN)
 /* don't include system-level gorp when generating doxygen files */
-#elif PMIX_ASSEMBLY_BUILTIN == PMIX_BUILTIN_SYNC
-#include "src/atomics/sys/sync_builtin/atomic.h"
 #elif PMIX_ASSEMBLY_BUILTIN == PMIX_BUILTIN_GCC
 #include "src/atomics/sys/gcc_builtin/atomic.h"
 #elif PMIX_ASSEMBLY_ARCH == PMIX_X86_64

